enclave_wrangler/dataset_upload.py

```diff
@@ -0,0 +1,355 @@
+
+    # Strip: remove the spaces in the beginning and the end of the name
+    df.columns.str.lstrip()
+    df.columns.str.rstrip()
+
+    # Float->Int: For some reason, was being read with .0's at the end.
+    for field in [x for x in integer_id_fields if x in list(df.columns)]:
+        df[field] = pd.to_numeric(df[field], errors='coerce')\
+            .astype('Int64')
+
+    return df
+
+
+def load_cache_if_valid(input_csv_folder_path) -> Union[pd.DataFrame, None]:
+    """Checks `enclave_codeset_id` and, if no empty vals, uses this file as cache."""
+    df: pd.DataFrame = _load_standardized_input_df(os.path.join(input_csv_folder_path, 'code_sets.csv'))
+    ids = list(df['enclave_codeset_id'])
+    valid = not any([x == '' for x in ids])
+
+    return df if valid else None
+
+
+def left_join_update(df, df2):
+    """Does a left join, but where column names are the same, keeps right value if exists, else left value."""
+    new_df = df.merge(
+        df2, how='left', left_on='internal_id', right_on='codeset_id')
+
+    xy_col_counts: Dict[str, int] = {}
+    for col in list(new_df.columns):
+        if any([col.endswith(x) for x in ['_x', '_y']]):
+            common_col = col[:-2]
+            if common_col not in xy_col_counts:
+                xy_col_counts[common_col] = 0
+            xy_col_counts[common_col] += 1
+
+    xy_cols: List[str] = [k for k, v in xy_col_counts.items() if v == 2]
+    for col in xy_cols:
+        new_df[col] = new_df[col + '_y'].fillna(new_df[col + '_x'])
+        new_df = new_df.drop([col + '_x', col + '_y'], axis=1)
+
+    return new_df
+
+
+def persist_to_db(code_sets_df) -> pd.DataFrame:
+    """Save updated items to persistence layer"""
+    # Vars
+    join_col = 'codeset_id'
+    update_cols = ['enclave_codeset_id', 'enclave_codeset_id_updated_at', 'concept_set_name']
+
+    # Read
+    persistence_csv_path = os.path.join(PROJECT_ROOT, 'data', 'cset.csv')
+    persistence_df = _load_standardized_input_df(persistence_csv_path)
+
+    # Subset
+    try:
+        code_sets_df_limited = code_sets_df[[join_col] + update_cols]
+    except KeyError:  # if KeyError, `join_col` is an index
+        code_sets_df_limited = code_sets_df[update_cols]
+    # Join
+    persistence_df_new = left_join_update(persistence_df, code_sets_df_limited)
+
+    # Save and return
+    persistence_df_new.to_csv(persistence_csv_path, index=False)
+    return persistence_df_new
+
+
+def run(input_csv_folder_path, use_cache=False):
+    """Main function"""
+    code_sets_df: Union[pd.DataFrame, None] = load_cache_if_valid(input_csv_folder_path) if use_cache else None
+    if code_sets_df is None:
+        code_sets_df: pd.DataFrame = post_to_enclave_and_update_code_sets_csv(input_csv_folder_path)
+    persist_to_db(code_sets_df)
+
+
+def cli():
+    """Command line interface for package.
+
+    Side Effects: Executes program."""
+    package_description = 'Tool for uploading to the Palantir Foundry enclave.'
```

Comment:
## `dataset_upload.py`
All of this stuff was in the `main.py` in the `valueset-tools/enclave_wrangler`.

## Enclave wrangler submodule?
I checked and there was a lot in `valueset-tools/enclave_wrangler` for this that did not make its way into the `termhub/enclave_wrangler`. I fear potential day where we will have to do a large batch of data management and it is hard to decide which `enclave_wrangler` to use. Maybe it will be time soon to put it in a `git submodule`?